cmd/fleetctl/vulnerability_data_stream.go

```diff
@@ -37,35 +41,82 @@ Downloads (if needed) the data streams that can be used by the Fleet server to p
 			if dir == "" {
 				return errors.New("No directory provided")
 			}
-
 			err := os.MkdirAll(dir, 0o700)
 			if err != nil {
 				return err
 			}
 
 			log(c, "[-] Downloading CPE database...")
-
 			dbPath := path.Join(dir, "cpe.sqlite")
 			client := fleethttp.NewClient()
 			err = vulnerabilities.SyncCPEDatabase(client, dbPath)
 			if err != nil {
 				return err
 			}
-
 			log(c, " Done\n")
 
 			log(c, "[-] Downloading CVE data streams...")
-
 			err = vulnerabilities.SyncCVEData(dir, config.FleetConfig{})
 			if err != nil {
 				return err
 			}
+			log(c, " Done\n")
 
+			log(c, "[-] Collecting Hosts OS info...")
+			fleetClient, err := clientFromCLI(c)
+			if err != nil {
+				return err
+			}
+			platforms, err := collectUniqPlatforms(fleetClient)
+			if err != nil {
+				return err
+			}
 			log(c, " Done\n")
 
-			log(c, "[+] Data streams successfully downloaded!\n")
+			log(c, "[-] Downloading Oval definitions...")
+			err = oval.Sync(client, dir, platforms)
+			if err != nil {
+				return err
+			}
+			log(c, " Done\n")
 
+			log(c, "[+] Data streams successfully downloaded!\n")
 			return nil
 		},
 	}
 }
+
+// Returns a set of supported oval platforms from all hosts
+func collectUniqPlatforms(client *service.Client) ([]oval.Platform, error) {
```

Comment:
*Re: Downloading all oval files.* I think that makes sense if on most FleetDM installations the number of distinct platforms is high, otherwise that will result in having to download/process hundreds of Mbs for nothing.

*Re: Fetching all the platforms this way...* Yeah I agree this is not ideal but, I couldn't find any other way to get the data I needed via the existing API. If it is the case that users run `fleetctl vulnerability-data-stream` often, then maybe we should look into creating an API end-point with the platform info.

Reply:
+1 to not have `fleetctl vulnerability-data-stream` command to get all hosts.
So I agree we should download all oval files in this first iteration.

PS: In the future, I would propose doing it the unix way, a separate command to "fetch all platforms" and then feed such output to the `fleetctl vulnerability-data-stream` command.

Reply:
:+1: